SOLIDITY_STYLE_GUIDE.md

```diff
@@ -0,0 +1,1504 @@
+# Style Guide
```

Comment:
Two changes we discussed: (a) giving credit to the base of this style guide and (b) explicitly casting with 0.